tcms/static/js/testrun_actions.js

```diff
@@ -244,63 +254,36 @@ Nitrate.TestRuns.AssignCase.on_load = function() {
   });
 };
 
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-function updateRunStatus(object_pk, value, callback) {
-  jQ.ajax({
-    'url': '/runs/case-run-update-status/',
-    'type': 'POST',
-    'data': {'object_pk': object_pk, 'status_id': value },
-    'success': function (data, textStatus, jqXHR) {
-      callback();
-    },
-    'error': function (jqXHR, textStatus, errorThrown) {
-      json_failure(jqXHR);
-    }
-  });
-}
+  const container = $('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+  addComment(executionId, caseId)
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  // Update the object when changing the status
+  jsonRPC('TestExecution.update', [executionId, {'status': executionStatusPk}], () => {
+    // Update the case run status icon
+    title.find('.execution_status_icon').each((_, icon) => {
+      executionStatuses.forEach(status => $(icon).removeClass(status.icon))
+
+      const execution = executionStatuses.find(tcs => tcs.id === parseInt(executionStatusPk))
+      $(icon).addClass(execution.icon)
+      $(icon).css('color', execution.color)
+    });
 
+    // Update related people
+    title.find('.link_tested_by').each(() => $(this).html(Nitrate.User.username));
     // Mark the case run to mine
     if (!title.is('.mine')) {
       title.addClass('mine');
     }
```

Comment:
hmmm, not sure what this is doing but maybe we don't need it ?

Reply:
https://github.com/kiwitcms/Kiwi/pull/1258/files#diff-f1f966298171ba7b8ab6c743bd88e634L31

we have this. I am not sure what it is doing, but it seems like some logic based on the class `mine`